packages/cli/src/WorkflowHelpers.ts

```diff
@@ -372,16 +373,24 @@ export async function replaceInvalidCredentials(workflow: WorkflowEntity): Promi
  * Get the IDs of the workflows that have been shared with the user.
  * Returns all IDs if user is global owner (see `whereClause`)
  */
-export async function getSharedWorkflowIds(user: User, roles?: string[]): Promise<string[]> {
+export async function getSharedWorkflowIds(user: User, roles?: RoleNames[]): Promise<string[]> {
+	const where: FindOptionsWhere<SharedWorkflow> = {};
+	if (user.globalRole.name !== 'owner') {
+		where.userId = user.id;
+	}
+	if (roles?.length) {
+		const roleIds = await Db.collections.Role.find({
+			select: ['id'],
+			where: { name: In(roles), scope: In(['workflow', 'global']) },
```

Comment:
do we really need `global` in here? 
```suggestion
			where: { name: In(roles), scope: 'workflow' },
```

Reply:
if for some reason we were looking for member,global i guess

Reply:
but that seems incorrect. every non-owner is a global member.